python-packages/presto_drf/presto_drf/parsers.py

```diff
@@ -0,0 +1,65 @@
+import json
+
+from allianceutils.util.camel_case import camel_to_underscore
+from allianceutils.util.camel_case import underscoreize
+from django.conf import settings
+from rest_framework.exceptions import ParseError
+from rest_framework.parsers import JSONParser
+from rest_framework.parsers import MultiPartParser
+
+
+class CamelCaseJSONParser(JSONParser):
+    def parse(self, stream, media_type=None, parser_context=None):
+        parser_context = parser_context or {}
+        encoding = parser_context.get("encoding", settings.DEFAULT_CHARSET)
+
+        try:
+            data = stream.read().decode(encoding)
+            return underscoreize(json.loads(data))
```

Comment:
One of the reasons the alliance utils `underscorize` was written was to be able to differentiate keys that are field names (and should be mangled) from keys that are mappings of a value1 => value2 (so value1 shouldn't be changed).

There doesn't seem to be any way to implement that here, even with subclassing. At minimum the underscorize() invocation should be through a method that you can override if needed

Reply:
Done